extensions/geode-modules/src/test/java/org/apache/geode/modules/util/ModuleFunctionsSecurityTest.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.modules.util;
+
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import org.apache.geode.cache.RegionShortcut;
+import org.apache.geode.cache.execute.FunctionService;
+import org.apache.geode.examples.SimpleSecurityManager;
+import org.apache.geode.test.junit.categories.IntegrationTest;
+import org.apache.geode.test.junit.rules.GfshCommandRule;
+import org.apache.geode.test.junit.rules.ServerStarterRule;
+
+@Category(IntegrationTest.class)
+public class ModuleFunctionsSecurityTest {
+
+  @ClassRule
+  public static ServerStarterRule server = new ServerStarterRule()
+      .withJMXManager()
+      .withSecurityManager(SimpleSecurityManager.class)
+      .withRegion(RegionShortcut.REPLICATE, "REPLICATE_1")
+      .withRegion(RegionShortcut.PARTITION, "PARTITION_1")
+      .withAutoStart();
+
+  @ClassRule
+  public static GfshCommandRule gfsh = new GfshCommandRule();
+
+  @BeforeClass
+  public static void setupClass() {
+    FunctionService.registerFunction(new BootstrappingFunction());
+    FunctionService.registerFunction(new CreateRegionFunction());
+    FunctionService.registerFunction(new RegionSizeFunction());
+    FunctionService.registerFunction(new TouchPartitionedRegionEntriesFunction());
+    FunctionService.registerFunction(new TouchReplicatedRegionEntriesFunction());
+  }
+
+  @After
+  public void teardown() throws Exception {
+    gfsh.disconnect();
+  }
+
+  @Test
+  public void testInvalidPermissionsForBootstrappingFunction() throws Exception {
```

Comment:
Another way to have the test auto connect with a specific user/password is to do this:

@Rule // instead of a classrule so that it connect/disconnect around each test
public GfshCommandRule gfsh = new GfshCommandRule(server.getJmxPort, PortType.JmxManager);

@Test
@ConnectionConfiguration(user = "dataWrite", password = "dataWrite")
public void test(){
    gfsh.executeAndAssertThat("execute function --id=test");
}